### PR TITLE
Pause changes, readme changes

### PR DESCRIPTION
**Pull request by insomniacUNDERSCORElemon**
- pausing moved from player to hud
- pause text, pause only allowed during play
- set new process mode (I did not upload this file before, so this is also a fix)
- readme changes, including compile instruction in dtc2d

### CODE REVIEW

File: dodge_the_creeps_2d/core/src/classes/hud.nim [hunk 1]
```diff
@@ -4,6 +4,11 @@ import gdextgen/classes/gdSceneTree
 import gdextgen/classes/gdLabel
 # import gdextgen/classes/gdButton
 import gdextgen/classes/gdTimer
+import gdextgen/classes/gdInput
+
+var
+  was_paused: bool = false
+  is_playing: bool = false
```
Comment (panno8M) [inline dodge_the_creeps_2d/core/src/classes/hud.nim]: It is more natural for these variables to exist as member variables of a class.

**Review by panno8M — CHANGES_REQUESTED** [review]
Considering the role of the class, it is preferable that the pause process be defined in the Main class. This is because this process involves the entire game.

The is_playing flag should be changed so that it is toggled in Main.new_game() and Main.game_over() functions. (This is more natural, and convenient for moving processing from Hud to Main.)

The Main class already has a reference to Hud, so porting should not be too difficult. You can use Hud.show_message() from Main.

File: dodge_the_creeps_2d/core/src/classes/hud.nim (hunk 1 above)
Comment (insomniacUNDERSCORElemon) [reply dodge_the_creeps_2d/core/src/classes/hud.nim]: I could do so, though needing `self.` for it makes it a bit uglier IMO.

**Comment by insomniacUNDERSCORElemon** [comment]
> Considering the role of the class, it is preferable that the pause process be defined in the Main class. This is because this process involves the entire game.

This would mean that the main scene would be the one set to `Always` process mode, and every node inherits process mode by default thus every root-level node (save for the hud node, as I did here) would need to be set to `Pausable`. This is a potential issue/annoyance when adding new root-level nodes to the main scene. In other words, it's a footgun. EDIT: Also another thing I didn't immediately think of, mob.tscn is also set to `Inherit`.

Due to this being Godot's pausing (not some custom state handling\*) I'm also not concerned with it being handled here. However I could see this being an issue in a multi-scene structure (where Hud is not always present) or ~~if classes did not still run code when their node is hidden~~ (EDIT: Hud itself is always visible). 

\*= `self.playing` is  only used to prevent pause text from overwriting a message that is already on the screen (doing so would be odd _and_ harder to deal with), not pausing (as a state) itself. This is why I do it on the end of the ready countdown rather than on a new game, I don't think pausing during the tiny countdown is needed anyways.

File: dodge_the_creeps_2d/core/src/classes/hud.nim (hunk 1 above)
Comment (insomniacUNDERSCORElemon) [reply dodge_the_creeps_2d/core/src/classes/hud.nim]: I've removed `was_paused` by handling visibility the same way as I did by inverting pausing.

EDIT: I guess the other _is_ relating to hud flow, so I changed that too.

**Comment by insomniacUNDERSCORElemon** [comment]
> it is preferable that the pause process be defined in the Main class

Given my comment about process mode above, if you still want me to port this to the Main class my solution would be to add another node with `Pausable` under Main, and then move all nodes (except Hud) in main.tscn to this to restore the pausable inheritance chain and make pausable/always difference clearer. 

That's assuming that doing so will only break node paths in `Main.nim`.

Y/N?

**Review by panno8M — APPROVED** [review]

**Comment by panno8M** [comment]
I understand the intent. The answer now is no.
I thought it would be beautiful if Hud were purely responsible for display processing and did not do any active processing itself. However, it is not a good idea to complicate the structure for the sake of beauty.

I'll merge in a little time based on the past faults

**Comment by insomniacUNDERSCORElemon** [comment]
> I thought it would be beautiful if Hud were purely responsible for display processing and did not do any active processing itself

For a more advanced project, it could be done handling input in a dedicated class (no children).

Being handled by UI (which logically shows on pause and should-not-be-paused, handles some input) isn't too bad, and thanks to the changes you requested I made it just about the cleanest/simplest that pausing (w/indicator) can be.

Godot's handling the pause, Hud just allows inverting it if Hud is in the 1 allowed state.

> I'll merge in a little time based on the past faults

Not sure if you mean you'll test it or if you're referring to quick merges, but either way is fair.

The only issue I see now is the editor opening to 3D mode instead of 2D, but I don't think that one is related to anything I did. (I think that may be a Godot issue related to node choice)

**Pull request merged**